src/bp/ui-admin/src/Pages/Workspace/CreateBotModal.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import _ from 'lodash'
+import { Button, Modal, ModalHeader, ModalBody, FormGroup, FormFeedback, Label, Input } from 'reactstrap'
+import { MdGroupAdd } from 'react-icons/lib/md'
+import Select from 'react-select'
+
+import api from '../../api'
+import { fetchBotTemplates, fetchBotCategories } from '../../reducers/bots'
+
+const defaultState = {
+  name: '',
+  template: null,
+  category: null,
+  error: null
+}
+
+class CreateBotModal extends Component {
+  state = { ...defaultState }
+
+  componentDidMount() {
+    if (!this.props.botCategoriesFetched) {
+      this.props.fetchBotCategories()
+    }
+    if (!this.props.templateFetched) {
+      this.props.fetchBotTemplates()
+    }
```

Comment:
The boolean isn't really needed, you could just check if bot categories/templates array is empty

Reply:
nop, bot template can be an empty array same goes for bot categories. in this case it would trigger an infinite loop